### PR TITLE
Regenerate contracts

### PR DESCRIPTION
**Pull request by martyall**
NOTE: Depends on https://github.com/f-o-a-m/purescript-web3-generator/pull/79, don't merge until that is in and we can use a stable ref

It turns out we weren't regenerating the purs contract modules when we made major changes to the generator, allowing some bugs to get through. This PR

- regenerates those modules for some really recent version of the generator
- puts all the necessary deps in package.json and packages.dhall to be able to build and use a local chanterelle to generate contract code 
- updates the solc dep to 0.6 (cause that's what's in chanterelle repo for some reason). Thus caused updates to the contract byte code we deploy

Long story short, the way we test and set up dependencies / generator tools for the tests is a total nightmare. I think we need to rethink things, maybe create a separate repo for these tests that uses git submodules to bring things in rather than dhall refs. In any case, this gets us back in working order to hopefully finish the 0.15 bump.

### CODE REVIEW

File: test/chanterelle/packages.dhall [hunk 1]
```diff
@@ -1,136 +1,126 @@
...
-      https://github.com/purescript/package-sets/releases/download/psc-0.14.2-20210629/packages.dhall sha256:534c490bb73cae75adb5a39871142fd8db5c2d74c90509797a80b8bb0d5c3f7b
+      https://raw.githubusercontent.com/f-o-a-m/chanterelle/v6.0.0/packages.dhall
+        sha256:0366ca8491b2d8e59ce763ae9af541b8d44145a5ad593fa9f6198a070199d5fc
+      with eth-core.repo = "https://github.com/f-o-a-m/purescript-eth-core"
+      with eth-core.version = "v8.0.0"
+      with web3 = ../../spago.dhall as Location
+      with web3-generator.repo = "https://github.com/f-o-a-m/purescript-web3-generator"
+      with web3-generator.version = "a7f81d900b7fc8f029b0474c5cb0affbae82e0c7"
 
-let overrides = {=}
```
Comment (martyall) [inline test/chanterelle/packages.dhall:4]: todo: remove
